lib/jekyll-feed/generator.rb

```diff
@@ -33,6 +33,23 @@ def config
       @config ||= @site.config["feed"] || {}
     end
 
+    # Returns the metadata for the collection in the argument
+    def data_for_collection(name)
+      @site.collections[name].metadata
+    end
+
+    # Returns the configured title for the collection if defined. Otherwise, it
+    # returns a fallback value of "Site Name | Collection"
+    def title(collection)
+      title = data_for_collection(collection)["title"]
+      title || "#{@site.config["name"]} | #{collection.capitalize}" unless title == "posts"
```

Comment:
I definitely agree with @ashmaroli here that this is a little hard to follow at first glance.

You could call this default piece `default_title(collection_name)` and restructure the method to early-return for `collection_name == posts`. Also, I'd update `collection` to be explicit that it's the `collection_name` or `collection_slug`, not the `Jekyll::Collection` object.

```ruby
def title(collection_name)
  return if collection_name == "posts"
  data_for_collection(collection)["title"] || default_title(collection)
end

def default_title(collection_name)
  "#{@site.config["name"]} | #{collection_name.capitalize}"
end
```